### ISSUE TITLE
[Feature Request] Multiple IPs for NAT Gateways

### ISSUE THREAD

**Issue by colstrom** [runs-on/runs-on#139]
When using NAT Gateways, egress traffic has a substantially reduced IP range, which leads to some issues that would be rarer on runners with public IPs becoming quite common on ones with private IPs, such as IP-based rate limiting (such as pulling images from DockerHub).

NAT Gateways can have multiple Elastic IPs attached, and egress traffic is spread across those more-or-less evenly, which mitigates this sort of problem at the cost of the additional IPs (which are relatively inexpensive).

Can we get an option to allocate additional IPs to each `PrivateSubnetGateway`? At least up to 8 per NGW (since that's what I'm using now), but more would be better, depending on how cumbersome it is to implement.

I have this functionality patched into my CloudFormation template, but I'm fairly confident that the way I've done so is... less than ideal (_"make it work, then make it right"_, right?):

```yaml
  PrivateSubnetGateway1:
    Condition: HasPrivateSubnet
    DependsOn:
    - ElasticIP1s1
    - ElasticIP1s2
    - ElasticIP1s3
    - ElasticIP1s4
    - ElasticIP1s5
    - ElasticIP1s6
    - ElasticIP1s7
    Type: AWS::EC2::NatGateway
    Properties:
      AllocationId: !GetAtt [ElasticIP1, AllocationId]
      SecondaryAllocationIds:
      - !GetAtt [ElasticIP1s1, AllocationId]
      - !GetAtt [ElasticIP1s2, AllocationId]
      - !GetAtt [ElasticIP1s3, AllocationId]
      - !GetAtt [ElasticIP1s4, AllocationId]
      - !GetAtt [ElasticIP1s5, AllocationId]
      - !GetAtt [ElasticIP1s6, AllocationId]
      - !GetAtt [ElasticIP1s7, AllocationId]
      SubnetId: !Ref PublicSubnet1
      Tags:
        - Key: "stack"
          Value: !Ref AWS::StackName

# and then further down...

  ElasticIP1:
    Condition: HasPrivateSubnet
    Type: AWS::EC2::EIP
    Properties:
      Domain: vpc
      Tags:
        - Key: "stack"
          Value: !Ref AWS::StackName

  ElasticIP2:
    Condition: HasPrivateSubnet
    Type: AWS::EC2::EIP
    Properties:
      Domain: vpc
      Tags:
        - Key: "stack"
          Value: !Ref AWS::StackName

  ElasticIP3:
    Condition: HasPrivateSubnet
    Type: AWS::EC2::EIP
    Properties:
      Domain: vpc
      Tags:
        - Key: "stack"
          Value: !Ref AWS::StackName

# etc etc etc
```

I don't (yet) understand CloudFormation templates well enough to do it better than that, but hopefully there's a more configuration-friendly construct available, like some sort of loop/iterator/range operator?

**Comment by crohr** [comment]
Thanks for the idea, and unfortunately I think your approach is the one I will have to implement. CF doesn't have loops. Could use custom resources or lambdas but this is probably too complex compared to this.

**Comment by colstrom** [comment]
Ah. That's... awkward, then.

Do you know if CF supports things like YAML anchors and aliases? Most parsers do, since it's part of the spec... but then there's things like GitHub Actions which doesn't for some reason, despite it being the [most popular open issue](https://github.com/actions/runner/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) (by number of 👍 reactions).

That doesn't exactly _solve_ the problem, but it would at least reduce some of the boilerplate...

**Comment by crohr** [comment]
Last time I tried it didn't support anchors. But it will be fine, just verbose. Might also just provide a generator at some point with common configs.

**Comment by crohr** [comment]
This will be available in v2.5.6. Note that by default NAT gateways are limited to 2 Elastic IPs, so I suppose you have asked for a quota increase first?
